### ISSUE TITLE
Net::Ping fails test on 5.28

### ISSUE THREAD

**Issue by demerphq** [Perl/perl5#21259]
Module:
Net::Ping

**Description**
Net::Ping test t/400_ping_syn.t seems to regularly fail test on 5.28 as follows:

```
 #   Failed test 'supposed to be down: http://yahoo.com./ [Timed out]'
  #   at t/400_ping_syn.t line 104.
  t/400_ping_syn.t ....... 
  Dubious, test returned 1 (wstat 256, 0x100)
  Failed 1/14 subtests 
  #          got: '1'
  #     expected: '0'
```

**Steps to Reproduce**
Push a PR to github, and then wait for the 5.28 dist tests to run.

See: https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783 for an example of this.

**Expected behavior**
The test should not fail.

**Perl configuration**

```
perl -V
  shell: sh -e {0}
  env:
    PERL_SKIP_TTY_TEST: 1
    CONTINUOUS_INTEGRATION: 1
    PERL_USE_UNSAFE_INC: 0
    AUTHOR_TESTING: 1
    AUTOMATED_TESTING: 1
    RELEASE_TESTING: 1
Summary of my perl5 (revision 5 version 24 subversion 4) configuration:
   
  Platform:
    osname=linux, osvers=4.19.0-14-cloud-amd64, archname=x86_64-linux-gnu
    uname='linux f585a0956915 4.19.0-14-cloud-amd64 #1 smp debian 4.19.171-2 (2021-01-[3](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:3)0) x86_6[4](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:4) gnulinux '
    config_args='-Darchname=x86_64-linux-gnu -Duse64bitall -Duseshrplib -Dvendorprefix=/usr/local -des'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat[5](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:5)005=undef
  Compiler:
    cc='cc', ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=[6](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:6)4 -D_FORTIFY_SOURCE=2',
    optimize='-O2',
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion='', gccversion='8.3.0', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=123456[7](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:7)[8](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:8), doublekind=3
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=[16](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:17), longdblkind=3
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/x86_64-linux-gnu/8/include-fixed /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib
    libs=-lpthread -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.28.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.28'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/local/lib/perl5/5.24.4/x86_64-linux-gnu/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options: HAS_TIMES PERLIO_LAYERS PERL_COPY_ON_WRITE
                        PERL_DONT_CREATE_GVSV
                        PERL_HASH_FUNC_ONE_AT_A_TIME_HARD PERL_MALLOC_WRAP
                        PERL_PRESERVE_IVUV USE_64_BIT_ALL USE_64_BIT_INT
                        USE_LARGE_FILES USE_LOCALE USE_LOCALE_COLLATE
                        USE_LOCALE_CTYPE USE_LOCALE_NUMERIC USE_LOCALE_TIME
                        USE_PERLIO USE_PERL_ATOF
  Built under linux
  Compiled at May 21 [20](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:21)[21](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:22) [23](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:24):05:03
  %ENV:
    PERL_SKIP_TTY_TEST="1"
    PERL_USE_UNSAFE_INC="0"
  @INC:
    /usr/local/lib/perl5/site_perl/5.[24](https://github.com/Perl/perl5/actions/runs/5567789042/jobs/10170051783#step:4:25).4/x86_64-linux-gnu
    /usr/local/lib/perl5/site_perl/5.24.4
    /usr/local/lib/perl5/vendor_perl/5.24.4/x86_64-linux-gnu
    /usr/local/lib/perl5/vendor_perl/5.24.4
    /usr/local/lib/perl5/5.24.4/x86_64-linux-gnu
    /usr/local/lib/perl5/5.24.4
```

**Comment by tonycoz** [comment]
It fails randomly in pinging yahoo, eg, from the 5.8 test for caef5a7dffddb63ccd397171f834d8f5122356ee:
```
t/300_ping_stream.t .... skipped: loopback tcp echo service is off (Connection refused)
  
  #   Failed test 'supposed to be down: http://yahoo.com./ [Timed out]'
  #   at t/400_ping_syn.t line 103.
  #          got: '1'
  #     expected: '0'
  # NOTE: Network connectivity will be required for all tests to pass.
  # Firewalls may also cause some tests to fail, so test it on a clear network.
  # Looks like you failed 1 test of 14.
  t/400_ping_syn.t ....... 
  Dubious, test returned 1 (wstat 256, 0x100)
  Failed 1/14 subtests 
```
From the MacOS test for cc54951f97bda0a3ef8cbe3552ad5784a9bb4392:
```
t/300_ping_stream.t .... skipped: loopback tcp echo service is off (Connection refused)
  
  #   Failed test 'supposed to be down: http://yahoo.com./ [Timed out]'
  #   at t/400_ping_syn.t line 104.
  #          got: '1'
  #     expected: '0'
  # NOTE: Network connectivity will be required for all tests to pass.
  # Firewalls may also cause some tests to fail, so test it on a clear network.
  # Looks like you failed 1 test of 14.
  t/400_ping_syn.t ....... 
  Dubious, test returned 1 (wstat 256, 0x100)
  ```
From the 5.8 test for 358f5bbe54cdef533446eee53fb5a7d073a721f6:

```
t/190_alarm.t .......... ok
  
  #   Failed test 'Can ping yahoo.com'
  #   at t/200_ping_tcp.t line 77.
  #          got: '0'
  #     expected: anything else
  # Looks like you failed 1 test of 13.
  t/200_ping_tcp.t ....... 
  Dubious, test returned 1 (wstat 256, 0x100)
  Failed 1/13 subtests 
```
so it's not just 5.24.

**Comment by xenu** [comment]
Fixed by #21278.
